lib/routes/account.js

```diff
@@ -1612,4 +1630,50 @@ module.exports = function (
   }
 
   return routes
+
+  function upsertDevice (request, sessionToken, deviceInfo) {
+    var operation = deviceInfo.id ? 'updateDevice' : 'createDevice'
+    return db[operation](sessionToken.uid, sessionToken.tokenId, deviceInfo)
+      .then(
+        function (device) {
+          if (operation === 'createDevice') {
+            log.event('device:create', request, {
+              uid: sessionToken.uid,
+              id: device.id,
+              type: device.type,
+              timestamp: device.createdAt
+            })
+          }
+          push.notifyDeviceConnected(sessionToken.uid, device.name, device.id.toString('hex'))
```

Comment:
> True, we should probably change this.

Cool, thanks. I'll update it in this PR.

Also, I still haven't looked at your Fennec patch, sorry, but I absolutely, definitely, positively, 100% promise and guarantee that I will do so tomorrow!